src/test/java/org/czekalski/petapiintegration/config/AnimalPageCaseFactoryTest.java

```diff
@@ -0,0 +1,149 @@
+package org.czekalski.petapiintegration.config;
+
+import org.czekalski.petapiintegration.apiclient.v1.dto.AnimalsListDto;
+import org.czekalski.petapiintegration.apiclient.v1.dto.PaginationDto;
+import org.czekalski.petapiintegration.config.animalfactory.AnimalPageCaseFactory;
+import org.czekalski.petapiintegration.resourceserver.apiresource.v1.dto.AnimalResourceDto;
+import org.czekalski.petapiintegration.resourceserver.apiresource.v1.dto.AnimalResourcesListDto;
+import org.junit.jupiter.api.Test;
+import org.springframework.data.domain.*;
+import org.springframework.hateoas.EntityModel;
+
+import java.util.Collections;
+import java.util.NoSuchElementException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class AnimalPageCaseFactoryTest {
+
+    private final AnimalPageCaseFactory animalPageCaseFactory = new AnimalPageCaseFactory();
+    private static final String CITY_ID = "Andover";
+    private static final String STATE_ID = "MA";
+
+
+    @Test
+    void AddLinksToFirstPage() {
```

Comment:
Ja osobiście lubię komentowanie given, when, then w JUnit, bo dodaje to czytelności

Reply:
poprawiono: 96619b9b46fa538703ca171c44c795010b4ffceb